src/main/java/com/amazon/services/UserServiceImpl.java

```diff
@@ -0,0 +1,65 @@
+package com.amazon.services;
+
+import com.amazon.dto.UserDto;
+import com.amazon.exception.InvalidUserException;
+import com.amazon.repository.IUserRepository;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@Service
+public class UserServiceImpl implements IUserService{
+
+    @Autowired
+    IUserRepository repository;
+
+    @Override
+    public UserDto saveUser(UserDto userDto) {
+        if(StringUtils.isBlank(userDto.getFirstName())){
+            throw new InvalidUserException("User.First name can not be empty");
+        }
+        if(StringUtils.isBlank(userDto.getPassword())){
+            throw new InvalidUserException("User.password can not be empty");
+        }
+
+        if(StringUtils.isBlank(userDto.getEmail()) && StringUtils.isBlank(userDto.getMobile()) ){
+            throw new InvalidUserException("User.Email or User.mobile both can not be empty");
+        }
+        if( userDto.getMobile().length() != 10){
+            throw new InvalidUserException("User.mobile should be 10 digit");
```

Comment:
change error message to "User.Mobile should contain 10 digits"

Reply:
done